libmscore/fret.h

```diff
@@ -197,6 +199,11 @@ class FretDiagram final : public Element {
       bool showNut() const        { return _showNut;    }
       void setShowNut(bool val)   { _showNut = val;     }
 
+      QString harmonyText() const { return _harmony ? _harmony->plainText() : QString(); }
+      void setHarmony(QString harmonyText);
+      void undoSetHarmony(QString harmonyText);
+      qreal numberWidth() const   { return xdiff;        }   // for harmony layout, a bit hacky
```

Comment:
Yes, it is :-).  It should be possible to avoid this using better bbox math - not just using width, but left & right as appropriate.  Instead of pos + (width/2), the actual midpoint is more like pos + (left+right)/2, if you know the thing isn't left-aligned.  That's off the top of my head, but I'll see about implementing it here.

Reply:
Actually, since the left portion is specifically what you want to ignore, what you really want to do is add right/2.